src/dev/flang/ast/Type.java

```diff
@@ -184,9 +176,9 @@ public String name()
    *
    * @param o
    */
-  public Type(SourcePosition pos, String n, List<AbstractType> g, AbstractType o)
+  public Type(String n, List<AbstractType> g, AbstractType o)
```

Comment:
IMHO a `Type` created by the parser should be the only type that has the original source code position. If a type in the source code is wrong (e.g., a spelling mistake), we want to point to that source code position. 
For all instances of AbstractType that are created later, there should not be any need for the souce code position.